parser/src/lib.rs

```diff
@@ -1,11 +1,19 @@
 #[macro_use]
 extern crate lalrpop_util;
 
-lalrpop_mod!(pub grammar);
-
 pub mod ast;
-pub mod error;
-pub mod lexer;
 
-#[cfg(test)]
-pub mod tests;
+mod error;
+use error::Error;
+
+mod lexer;
+use lexer::{Lexer, Token};
+
+mod parser;
+use crate::parser::SourceParser;
+
+/// Parses the provided source and returns the AST.
+pub fn parse(source: &str) -> Result<ast::Source, lalrpop_util::ParseError<usize, Token, Error>> {
```

Comment:
nit: Should we import module items instead of using full paths for better readability?

Reply:
No that makes sense. Thanks for the clarification.